Library/Homebrew/dev-cmd/audit.rb

```diff
@@ -49,20 +49,34 @@
 require "date"
 require "missing_formula"
 require "digest"
+require "cli_parser"
 
 module Homebrew
   module_function
 
   def audit
-    inject_dump_stats!(FormulaAuditor, /^audit_/) if ARGV.switch? "D"
+    args = Homebrew::CLI::Parser.new do
+      switch      "--strict",             "strict"
+      switch      "--online",             "online"
+      switch      "--new-formula",        "new-formula"
+      switch      "--fix",                "fix"
+      switch      "--display-cop-names",  "display cop names"
+      switch      "--display-filename",   "display file names"
+      switch      "-D",                   "Activates debugging and profiling"
+      comma_array "--only",               "only"
+      comma_array "--except",             "except"
+      comma_array "--only-cops",          "only cops"
+      comma_array "--except-cops",        "except cops"
+    end.parse
+
     Homebrew.auditing = true
+    inject_dump_stats!(FormulaAuditor, /^audit_/) if args.D?
```

Comment:
`args.D?` feels a little odd. Shall we give it a long name for which `-D` on the command line can be a shorthand? `--audit-debug` or something?

Reply:
Yeah. I've updated the code to support both short and long switches now, and added corresponding test.